lib/ui/painting/multi_frame_codec.cc

```diff
@@ -12,16 +12,18 @@
 
 namespace flutter {
 
-MultiFrameCodec::MultiFrameCodec(
-    std::shared_ptr<SkCodecImageGenerator> generator)
+MultiFrameCodec::MultiFrameCodec(std::shared_ptr<ImageGenerator> generator)
     : state_(new State(std::move(generator))) {}
 
 MultiFrameCodec::~MultiFrameCodec() = default;
 
-MultiFrameCodec::State::State(std::shared_ptr<SkCodecImageGenerator> generator)
+MultiFrameCodec::State::State(std::shared_ptr<ImageGenerator> generator)
     : generator_(std::move(generator)),
-      frameCount_(generator_->getFrameCount()),
-      repetitionCount_(generator_->getRepetitionCount()),
+      frameCount_(generator_->GetFrameCount()),
+      repetitionCount_(generator_->GetPlayCount() ==
+                               ImageGenerator::kInfinitePlayCount
+                           ? -1
+                           : generator_->GetPlayCount() - 1),
```

Comment:
Yeah, it seems like it'd be nice if we could just use `GetPlayCount` without translating here...